### DISCUSSION TITLE
After clear it's broken in Xcode?

### DISCUSSION THREAD

**Discussion by coooliang** [ccache/ccache#284] [General]
I use ccache in xcode,when I exec cleanup and clear command line it's broken,it's doesn't work??
ccache -s    always zero
cache size no longer change

coooliang:~ su$ ccache -s
cache directory                     /Users/su/.ccache
primary config                      /Users/su/.ccache/ccache.conf
secondary config      (readonly)    /usr/local/Cellar/ccache/3.4.2/etc/ccache.conf
stats zero time                     Mon Jul 16 15:43:30 2018
cache hit (direct)                     0
cache hit (preprocessed)               0
cache miss                             0
cache hit rate                      0.00 %
couldn't find the compiler             3
cleanups performed                     0
files in cache                         0
cache size                           0.0 kB
max cache size                       5.0 GB

**Comment by coooliang** [comment]
I'm add User-Defined in Xcode content is :

#!/bin/sh
if type -p ccache >/dev/null 2>&1; then
  export CCACHE_MAXSIZE=10G
  export CCACHE_CPP2=true
  export CCACHE_HARDLINK=true
  export CCACHE_SLOPPINESS=file_macro,time_macros,include_file_mtime,include_file_ctime,file_stat_matches
  export CCACHE_LOGFILE='~/Desktop/CCache.log'
  exec ccache /usr/bin/clang++ "$@"
else
  exec clang++ "$@"
fi

**Comment by afbjorklund** [comment]
What does the log file say ? (maybe "ccache" isn't in the PATH)

**Comment by dashchak** [comment]
Issue is reproduced in Xcode 10 betas, if I remove project DerivedData folder and start compiling.
ccache is in the $PATH as a prefix part. However, 
$ which ccache 
returns empty string.
In order to fix it, replace "ccache" calls with absolute path: "/usr/local/bin/ccache" 

P.S. For some reason Xcode's
$ echo $PATH 
returns incomplete PATH after cleanup:
/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin

**Comment by jrosdahl** [comment]
I have no idea about how to set up Xcode. I suggest asking in another forum.

One thing though, this does not work like you think:

    export CCACHE_LOGFILE='~/Desktop/CCache.log'

The `~` won't be expanded by ccache and you won't get any log file. Change it to something like

    export CCACHE_LOGFILE=$HOME/Desktop/CCache.log

instead. (Note that fixing this won't fix your original issue, it's just something I happened to notice.)

**Comment by S1GSEGV** [comment]
try add:

#!/bin/sh
export PATH=$PATH:/usr/local/bin
